TiCDC panic when truncate table

### How to reproduce
make integration_test CASE=multi_source

### Log
```
[2025/01/07 18:21:54.197 +08:00] [ERROR] [multi_version.go:106] ["table info deleted"] [ts=455146290702188556] [tableID=174] [infos="[{}]"] [deleteVersion=455146290702188555]
[2025/01/07 18:21:54.197 +08:00] [PANIC] [event_broker.go:541] ["get table info failed, unknown reason"] [error="table info deleted 174"] [stack="github.com/pingcap/ticdc/pkg/eventservice.(*eventBroker).doScan\n\tgithub.com/pingcap/ticdc/pkg/eventservice/event_broker.go:541\ngithub.com/pingcap/ticdc/pkg/eventservice.(*eventBroker).runScanWorker\n\tgithub.com/pingcap/ticdc/pkg/eventservice/event_broker.go:220\ngithub.com/pingcap/ticdc/pkg/eventservice.newEventBroker.func1\n\tgithub.com/pingcap/ticdc/pkg/eventservice/event_broker.go:138\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.8.0/errgroup/errgroup.go:78"]
```

--- a/logservice/schemastore/multi_version.go
+++ b/logservice/schemastore/multi_version.go
@@ -15,7 +15,6 @@ package schemastore
 
 import (
 	"errors"
-	"fmt"
 	"math"
 	"sort"
 	"sync"
@@ -93,12 +92,10 @@ func (v *versionedTableInfoStore) waitTableInfoInitialized() {
 	<-v.readyToRead
 }
 
-type TableDeletedError struct {
-	Msg string
-}
+type TableDeletedError struct{}
 
 func (e *TableDeletedError) Error() string {
-	return e.Msg
+	return "table deleted"
 }
 
 // return the table info with the largest version <= ts
@@ -116,7 +113,7 @@ func (v *versionedTableInfoStore) getTableInfo(ts uint64) (*common.TableInfo, er
 			zap.Any("tableID", v.tableID),
 			zap.Any("infos", v.infos),
 			zap.Any("deleteVersion", v.deleteVersion))
-		return nil, &TableDeletedError{Msg: fmt.Sprintf("table info deleted %d", v.tableID)}
+		return nil, &TableDeletedError{}
 	}
 
 	target := sort.Search(len(v.infos), func(i int) bool {

--- a/logservice/schemastore/multi_version.go
+++ b/logservice/schemastore/multi_version.go
@@ -93,6 +93,14 @@ func (v *versionedTableInfoStore) waitTableInfoInitialized() {
 	<-v.readyToRead
 }
 
+type TableDeletedError struct {
+	Msg string
+}
+
+func (e *TableDeletedError) Error() string {
+	return e.Msg
+}
+
 // return the table info with the largest version <= ts
 func (v *versionedTableInfoStore) getTableInfo(ts uint64) (*common.TableInfo, error) {
 	v.mu.Lock()
@@ -103,12 +111,12 @@ func (v *versionedTableInfoStore) getTableInfo(ts uint64) (*common.TableInfo, er
 	}
 
 	if ts >= v.deleteVersion {
-		log.Error("table info deleted",
+		log.Warn("table info deleted",
 			zap.Any("ts", ts),
 			zap.Any("tableID", v.tableID),
 			zap.Any("infos", v.infos),
 			zap.Any("deleteVersion", v.deleteVersion))
-		return nil, fmt.Errorf("table info deleted %d", v.tableID)
+		return nil, &TableDeletedError{Msg: fmt.Sprintf("table info deleted %d", v.tableID)}
 	}
 
 	target := sort.Search(len(v.infos), func(i int) bool {

--- a/pkg/eventservice/event_broker.go
+++ b/pkg/eventservice/event_broker.go
@@ -15,6 +15,7 @@ package eventservice
 
 import (
 	"context"
+	"errors"
 	"strconv"
 	"sync"
 	"time"
@@ -548,6 +549,11 @@ func (c *eventBroker) doScan(ctx context.Context, task scanTask) {
 			if err != nil {
 				if task.isRemoved.Load() {
 					log.Warn("get table info failed, since the dispatcher is removed", zap.Error(err))
+					return
+				} else if errors.Is(err, &schemastore.TableDeletedError{}) {
+					// After a table is truncated, it is possible to receive more dml events, just ignore is ok.
+					// TODO: tables may be deleted in many ways, we need to check if it is safe to ignore later dmls in all cases.
+					log.Warn("get table info failed, since the table is deleted", zap.Error(err))
 					return
 				}
 				log.Panic("get table info failed, unknown reason", zap.Error(err))

